### PR TITLE
clean up the build a bit. We no longer need the separate build_libtorch entrypoint

### PR DESCRIPTION
**Pull request by anderspapitto**
the new entrypoint is `./tools/build_pytorch_libs.sh caffe2`

this will also speed up CI builds a bit, since we will no longer be compiling all of libtorch twice

### CODE REVIEW

File: .jenkins/pytorch/build.sh
Comment (goldsborough) [inline .jenkins/pytorch/build.sh]: We don't need this anymore because the normal `setup.py` will build this anyway, correct?

File: .jenkins/pytorch/macos-build.sh
Comment (goldsborough) [inline .jenkins/pytorch/macos-build.sh]: I guess the non-hacky way would be to add a flag to our build to create this install directory in the torch package and install the binaries there. Do you think it would be a lot of effort?

File: tools/cpp_build/build_caffe2.sh
Comment (goldsborough) [inline tools/cpp_build/build_caffe2.sh]: Will `USE_OPENMP` still work?

File: tools/cpp_build/build_common.sh
Comment (goldsborough) [inline tools/cpp_build/build_common.sh]: Will debug remain the default?

File: tools/cpp_build/build_common.sh
Comment (goldsborough) [inline tools/cpp_build/build_common.sh]: Will ninja still be used if available?

File: tools/cpp_build/build_common.sh
Comment (goldsborough) [inline tools/cpp_build/build_common.sh]: Will the number of jobs still be the number of cores?

**Review by goldsborough — COMMENTED** [review]
Mostly LGTM, just a lot of questions of how my old config will translate to the brave new world

File: .jenkins/pytorch/build.sh
Comment (anderspapitto) [reply .jenkins/pytorch/build.sh]: yes

File: .jenkins/pytorch/macos-build.sh
Comment (anderspapitto) [reply .jenkins/pytorch/macos-build.sh]: I can give it a try. It's really annoying to iterate on the macos CI environment though, so if it's a big hassle I may punt on this. I really don't think it's all that bad

File: tools/cpp_build/build_caffe2.sh
Comment (anderspapitto) [reply tools/cpp_build/build_caffe2.sh]: yes, though when you're using the build_pytorch_libs it will look like

```
EXTRA_CAFFE2_CMAKE_FLAGS="USE_OPENMP=1" bash tools/build_pytorch_libs.sh ...
```

which is a bit ugly, but it does work and we are already separately working on making build_pytorch_libs "nicer"

File: .jenkins/pytorch/macos-build.sh
Comment (anderspapitto) [reply .jenkins/pytorch/macos-build.sh]: I've started working on this in a branch here https://github.com/pytorch/pytorch/pull/10136, but I'd like to get this diff pushed in the meantime without blocking on that

Comment (ezyang) [reply .jenkins/pytorch/macos-build.sh]: It doesn't make any sense to write cmake to install test binaries. But maybe what I would appreciate is a little script that handles this, so we can put this in a centralized area. Actually, for other reasons, I'd like this binaries to show up in the torch directory for Linux too (my CircleCI config for PyTorch uses 7z rather than docker push).

File: .jenkins/pytorch/build.sh
Comment (ezyang) [inline .jenkins/pytorch/build.sh]: Wait, why do you need this directory? On my build it weighs 700M, a good 30% of the overall 2.6G that's the build directory. And if we installed PyTorch, we should already have these libraries skulking around elsewhere. Is the rpath not working or something?

**Comment by goldsborough** [comment]
What is the status here? Can we make this happen soon?

**Comment by anderspapitto** [comment]
@goldsborough if you want to accept I'm ready to push now. But in the meantime i'm looking at some of the cleanup comments

**Review by goldsborough — APPROVED** [review]
@ezyang still had some questions, then he can give you codeowner approval

File: .jenkins/pytorch/build.sh
Comment (ezyang) [reply .jenkins/pytorch/build.sh]: > It maintains the behavior from before this diff that the tests are run out of their build directory

But no one actually cares about that. The actual functional requirements are:

1. The tests in CI need to find the libraries, somehow. (E.g., you could just use `LD_LIBRARY_PATH` to fix the path situation)
2. When I, a random developer, run tests from the build directory, they ought to work (without me having to `LD_LIBRARY_PATH` explicitly)
3. We should not be shipping multiple copies of the same libraries and binaries in the Docker image when we push from CPU to GPU. (But if you want to show that this doesn't actually make a difference, e.g., in time to push, then we can drop this requirement.)

**Comment by goldsborough** [comment]
I kind of expected you to wait for @ezyang's approval

**Pull request closed without merging**
